test/baseVisitors.test.ts

```diff
@@ -0,0 +1,333 @@
+  //   TsExpressionWithTypeArguments<S>(n: swc.TsExpressionWithTypeArguments, st: S, cb: Callback<S>) {
+  //     cb(n.expression, st)
+
+  //     if (n.typeArguments) {
+  //       cb(n.typeArguments, st)
+  //     }
+  //   }
+  //   TsInterfaceDeclaration<S>(n: swc.TsInterfaceDeclaration, st: S, cb: Callback<S>) {
+  //     cb(n.id, st)
+  //     cb(n.body, st)
+
+  //     for (const ext of n.extends) {
+  //       cb(ext, st)
+  //     }
+
+  //     if (n.typeParams) {
+  //       cb(n.typeParams, st)
+  //     }
+  //   }
+  //   TsInterfaceBody<S>(n: swc.TsInterfaceBody, st: S, cb: Callback<S>) {
+  //     for (const ele of n.body) {
+  //       cb(ele, st)
+  //     }
+  //   }
+  //   TsKeywordType = ignore
+  //   TsPropertySignature<S>(n: swc.TsPropertySignature, st: S, cb: Callback<S>) {
+  //     for (const param of n.params) {
+  //       cb(param, st)
+  //     }
+
+  //     cb(n.key, st)
+
+  //     if (n.typeAnnotation) {
+  //       cb(n.typeAnnotation, st)
+  //     }
+
+  //     if (n.typeParams) {
+  //       cb(n.typeParams, st)
+  //     }
+  //   }
+  //   TsAsExpression<S>(n: swc.TsAsExpression, st: S, cb: Callback<S>) {
+  //     cb(n.expression, st)
+  //     cb(n.typeAnnotation, st)
+  //   }
+  //   TsCallSignatureDeclaration<S>(n: swc.TsCallSignatureDeclaration, st: S, cb: Callback<S>) {
+  //     for (const param of n.params) {
+  //       cb(param, st)
+  //     }
+  //     if (n.typeParams) {
+  //       cb(n.typeParams, st)
+  //     }
+  //     if (n.typeAnnotation) {
+  //       cb(n.typeAnnotation, st)
+  //     }
+  //   }
+  {
+    type: 'TsExpressionWithTypeArguments',
+    code: 'type foo<T> = Array<T>',
+    mock: mock.method(baseVisitor, 'TsExpressionWithTypeArguments'),
+    skip: true,
+  },
+  {
+    type: 'TsInterfaceDeclaration',
+    code: 'interface Foo {}',
+    mock: mock.method(baseVisitor, 'TsInterfaceDeclaration'),
+  },
+  { type: 'TsInterfaceBody', code: 'interface Foo {}', mock: mock.method(baseVisitor, 'TsInterfaceBody') },
+  { type: 'TsKeywordType', code: 'type foo = string', mock: mock.method(baseVisitor, 'TsKeywordType') },
+  {
+    type: 'TsPropertySignature',
+    code: 'type foo = {a: string}',
+    mock: mock.method(baseVisitor, 'TsPropertySignature'),
+  },
+  { type: 'TsAsExpression', code: 'a as b', mock: mock.method(baseVisitor, 'TsAsExpression') },
+  {
+    type: 'TsCallSignatureDeclaration',
+    code: 'type foo = () => void',
+    mock: mock.method(baseVisitor, 'TsCallSignatureDeclaration'),
+  },
+]
+
+for (const options of tests) {
+  if (options.skip) {
+    continue
+  }
+  test(`should call ${options.type}`, () => {
+    options.mock.mock.resetCalls()
+    expect(given(options)).toHaveBeenCalledTimes(1)
+  })
+}
+
+function given(options: (typeof tests)[number]) {
+  const ast = reparseSync(options.code, 'ts')
+  if (options.debug) {
+    console.log(JSON.stringify(ast, null, 2))
+  }
+  simple(ast, {}, baseVisitor)
```

Comment:
I missed last time that `walk.ts` should add `Node` to the [type import](https://github.com/morganney/swc-walk/blob/main/src/walk.ts#L4) so that the underlying type is from an SWC Node and not a DOM Node.